angel_system/eval/visualization.py

```diff
@@ -4,74 +4,150 @@
+                # Ground truth
+                # ============================
+                # Bar plt to show bars where the "true" time ranges are for the activity.
+                xs_bars = [p - min_start_time for p in gt_ranges['start'].tolist()]
+                ys_gt_regions = [1] * gt_ranges.shape[0]
+                bar_widths = [p for p in gt_ranges['end']-gt_ranges['start']]
+                ax.bar(xs_bars, ys_gt_regions, width=bar_widths, align="edge", color="lightgreen", label="Ground truth")
+
+                if custom_range:
+                    assert len(custom_range) == 2, "Assuming only two float values for custom range"
+                    xs_bars2 = [custom_range[0]]
+                    ys_height = [1.025] #[0.1]
+                    bar_widths2 = [custom_range[1]-custom_range[0]]
+                    ys_bottom = [0] #[1.01]
+                    # TODO: Make this something that is added be clicking?
+                    ax.bar(xs_bars2, ys_height,
+                        width=bar_widths2, bottom=ys_bottom, align="edge",
+                        color=custom_range_color, alpha=0.5)
+
+                # ============================
+                # Detections
+                # ============================
+                xs2_bars = [p - min_start_time for p in det_ranges['start'].tolist()]
+                ys2_det_regions = [p for p in det_ranges['conf']]
+                bar_widths2 = [p for p in det_ranges['end']-det_ranges['start']]
+                ax.bar(xs2_bars, ys2_det_regions, width=bar_widths2, align="edge", edgecolor="blue", fill=False, label="Detections")
+
+                ax.legend(loc="upper right")
+                ax.plot
+
+                # ============================
+                # Save
+                # ============================
+                #plt.show()
+                activity_plot_dir = Path(os.path.join(self.output_dir, "activities"))
+                activity_plot_dir.mkdir(parents=True, exist_ok=True)
+                fig.savefig(f"{str(activity_plot_dir)}/{label.replace(' ', '_')}.png")
+            else:
+                log.warning(f"No detections/gt found for \"{label}\"")
+                
+    def plot_pr_curve(self, detect_intersection_thr=0.1):
+        """
+        Plot the PR curve for each label
+
+        :param detect_intersection_thr: detection intersection threshold
+        """
+        # ============================
+        # Setup figure
+        # ============================
+        fig, ax = plt.subplots(figsize=(7, 8))
+
+        ax.set_xlim([0.0, 1.0])
+        ax.set_ylim([0.0, 1.05])
+        ax.set_title("Precision vs. Recall")
+
+        colors = plt.cm.rainbow(np.linspace(0, 1, len(self.labels)))
+        
+        # ============================
+        # Add F1 score 
+        # ============================
+        fscores = np.linspace(0.2, 0.8, num=4)
+        for f_score in fscores:
+            x = np.linspace(0.01, 1)
```

Comment:
The plot as-is shows the f1=0.2 line to have a floating start:
![PR](https://user-images.githubusercontent.com/735270/189994699-64805eb0-52b2-4433-9776-5d3fa8669e73.png)

`0.001` results in a completed topographic line for the area covered. 
![PR](https://user-images.githubusercontent.com/735270/189994785-f9c0197c-a8fb-4413-9371-e85f7a674adc.png)
```suggestion
            x = np.linspace(0.001, 1)
```